ngraph_creator/operations/src/Strided_Slice.cpp

```diff
@@ -0,0 +1,108 @@
+#include <Strided_Slice.hpp>
+#define LOG_TAG "Strided_Slice"
+
+namespace android {
+namespace hardware {
+namespace neuralnetworks {
+namespace nnhal {
+
+Strided_Slice::Strided_Slice(int operationIndex) : OperationsBase(operationIndex) {
+    mDefaultOutputIndex = sModelInfo->getOperationOutput(mNnapiOperationIndex, 0);
+}
+
+bool Strided_Slice::validate() {
+    // Check input rank
+    const int64_t inputRank = getInputOperandDimensions(0).size();
+    if (inputRank > 4) {
+        ALOGE("%s Invalid input dimensions size!", __func__);
+        return false;
+    }
+
+    // TODO: Add Support for all_tensors_as_inputs
+    auto& begins_OperandIndex = sModelInfo->getOperationInput(mNnapiOperationIndex, 1);
+    if (!sModelInfo->isOperandLifeTimeConst(begins_OperandIndex)) {
+        ALOGE("%s Only Constant dimensions supported now", __func__);
+        return false;
+    }
+
+    auto ends_OperandIndex = sModelInfo->getOperationInput(mNnapiOperationIndex, 2);
+    if (!sModelInfo->isOperandLifeTimeConst(ends_OperandIndex)) {
+        ALOGE("%s Only Constant dimensions supported now", __func__);
+        return false;
+    }
+
+    auto& strides_OperandIndex = sModelInfo->getOperationInput(mNnapiOperationIndex, 3);
+    if (!sModelInfo->isOperandLifeTimeConst(strides_OperandIndex)) {
+        ALOGE("%s Only Constant dimensions supported now", __func__);
+        return false;
+    }
+
+    auto shrink_axis_mask = sModelInfo->ParseOperationInput<int32_t>(mNnapiOperationIndex, 6);
```

Comment:
Nit : Do you think when we have many indices in the operation we can avoid magic numbers by defining them?